NitroxPatcher/Patches/Client/CyclopsHelmHUDManager_Patch.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Harmony;
+using NitroxModel.Helper;
+using System;
+using System.Reflection;
+
+namespace NitroxPatcher.Patches.Client
+{
+    public class CyclopsHelmHUDManager_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(CyclopsHelmHUDManager);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("StopPiloting", BindingFlags.Public | BindingFlags.Instance);
+        public static readonly MethodInfo TARGET_METHOD_Start = TARGET_CLASS.GetMethod("Start", BindingFlags.NonPublic | BindingFlags.Instance);
+        public static readonly MethodInfo TARGET_METHOD_Update = TARGET_CLASS.GetMethod("Update", BindingFlags.NonPublic | BindingFlags.Instance);
+
+        public static void Start(CyclopsHelmHUDManager __instance)
+        {
+            __instance.ReflectionSet("hudActive", true);
+            if (__instance.motorMode.engineOn)
+            {
+                __instance.engineToggleAnimator.SetTrigger("EngineOn");
+            }
+            else
+            {
+                __instance.engineToggleAnimator.SetTrigger("EngineOff");
+            }
+        }
+
+        public static void Postfix(CyclopsHelmHUDManager __instance)
+        {
+            __instance.ReflectionSet("hudActive", true);
+        }
+
+        public static void Update(CyclopsHelmHUDManager __instance)
+        {
+            if (!__instance.hornObject.activeSelf && (bool)__instance.ReflectionGet("hudActive"))
```

Comment:
Same as below: isn't checking `hornObject.activeSelf` redundant?

Reply:
No, "hudActive" is almost always on so I have to check with the hornObjct if the Player is piloting or not.

Reply:
Indeed I see now: you use `hornObject.activeSelf` to check if the game itself wants the helmhud to be enabled, and `hudActive` to keep the hud active and updated (synced) in case the player is not 'using' it.

I strongly advice to add some comments explaining the reason and logic in patches: What the game does, why it needs a patch, and why you're patching it like such. It helps the review process (as right now I need to search through the game code), but more importantly it helps our future selves figure out what's going on when the game code changes and the patch does not work anymore.